### PR TITLE
Implement Explicit Frozen Dataclass

### PR DESCRIPTION
**Pull request by lionellloh**
## Summary
Implemented Explicit Frozen Dataclass

## Test Plan
Ran `tox -autofix` 
```
autofix run-test-pre: PYTHONHASHSEED='2908652032'
autofix run-test: commands[0] | isort -q .
autofix run-test: commands[1] | black .
All done! ✨ 🍰 ✨
95 files left unchanged.
_______________________________________________________________ summary ________________________________________________________________
  autofix: commands succeeded
  congratulations :)
```

### CODE REVIEW

File: fixit/rules/explicit_frozen_dataclass.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                ),
+            ):
+                if isinstance(decorator, cst.Call):
+                    func = decorator.func
+                    args = decorator.args
+                else:  # decorator is either cst.Name or cst.Attribute
+                    args = ()
+                    func = decorator
+
+                # pyre-fixme[29]: `typing.Union[typing.Callable(tuple.__iter__)[[], typing.Iterator[Variable[_T_co](covariant)]], typing.Callable(typing.Sequence.__iter__)[[], typing.Iterator[cst._nodes.expression.Arg]]]` is not a function.
```
Comment (lionellloh) [inline fixit/rules/explicit_frozen_dataclass.py:246]: Not sure if this is a proper/valid way to deal with the pyre error. But it seems like there might be some upstream blockers with pyre based on what I found internally.

File: fixit/rules/explicit_frozen_dataclass.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+class ExplicitFrozenDataclassRule(CstLintRule):
+    """
+    Encourages the use of frozen dataclass objects by telling users to specify the
+    kwarg.
+
+    Without this lint rule, most users of dataclass won't know to use the kwarg, and
+    may unintentionally end up with mutable objects.
+    """
+
+    ONCALL_SHORTNAME = "instagram_server_framework"
```
Comment (jimmylai) [inline fixit/rules/explicit_frozen_dataclass.py:23]: This line is not needed.

File: fixit/rules/explicit_frozen_dataclass.py (hunk 1 above)
Comment (jimmylai) [reply fixit/rules/explicit_frozen_dataclass.py:246]: Did pyre complain about the `any()` is not a function?

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
 flake8>=3.8.1
-libcst>=0.3.10
+libcst==0.3.13
```
Comment (jimmylai) [inline requirements.txt:2]: LibCST 0.3.15 should fixed the issue. So this change is no longer needed.

**Review by jimmylai — APPROVED** [review]
Overall LGTM. Some minor changes are needed.

File: fixit/rules/explicit_frozen_dataclass.py (hunk 1 above)
Comment (lionellloh) [reply fixit/rules/explicit_frozen_dataclass.py:246]: Oh thanks for the catch!

File: fixit/rules/explicit_frozen_dataclass.py (hunk 2 above)
Comment (lionellloh) [reply fixit/rules/explicit_frozen_dataclass.py:23]: Ah, thanks for the catch

Comment (lionellloh) [reply fixit/rules/explicit_frozen_dataclass.py]: Regarding > Did pyre complain about the any() is not a function?
It didn't state if it was `any()` that was violating. The full error is below. 
```fixit/rules/explicit_frozen_dataclass.py:245:79 Call error [29]: `typing.Union[typing.Callable(tuple.__iter__)[[], typing.Iterator[Variable[_T_co](covariant)]], typing.Callable(typing.Sequence.__iter__)[[], typing.Iterator[cst._nodes.expression.Arg]]]` is not a function.```

**Comment by lionellloh** [comment]
Thanks a lot @jimmylai for your patience and mentorship!

**Comment by jimmylai** [comment]
> Thanks a lot @jimmylai for your patience and mentorship!

You're welcome! Thanks for contributing, @lionellloh .

**Pull request merged**